tests/ext/test_callbackqueryhandler.py

```diff
@@ -242,9 +242,23 @@ def test_game_pattern(self, callback_query):
         callback_query.callback_query.game_short_name = "this is a short game name"
         assert not handler.check_update(callback_query)
 
-        callback_query.callback_query.game_short_name = object()
-        assert not handler.check_update(callback_query)
-
         callback_query.callback_query.data = "something"
         handler = CallbackQueryHandler(self.callback_basic, game_pattern="")
         assert not handler.check_update(callback_query)
+
+    @pytest.mark.parametrize(
+        ("data", "pattern", "game_short_name", "game_pattern", "expected_result"),
+        [
+           ("data", "data", None, "game_data", True),
+           (None, "data", "game_data", ".*data", True),
```

Comment:
https://github.com/python-telegram-bot/python-telegram-bot/pull/4353/files/a17b042796a345406c60f5b583e3fc9ba5d030bc..6fd56b59d07ff1e008a50f7012e47d8f65f737b1#r1674145424

Basically we have for combinations of `pattern` and `game_pattern`  as listed in https://github.com/python-telegram-bot/python-telegram-bot/pull/4353#issue-2393926075. for each of those, we want to check the three types of callback queries (data, game_short_name, none of both - the last admittedly only for completeness sake as this case doesn't exist in the Bot API currently). So we'll need 4×3 = 12 combinations here.

BTW please install the pre-commit hooks locally as explained in the contrib guide (run `pre-commit install`). that way git just won't let you commit before these checks pass ;) you can also install the used checks (`black`, `ruff`, `mypy`, …) from the pypi and run them in your command line. running `black .` will just auto-format everything for you.

Reply:
Thank you for the pre-commit thing. 

I have [updated the test cases](https://github.com/python-telegram-bot/python-telegram-bot/blob/57db295265a9e38ecdfed93807bf159bbd29ac6a/tests/ext/test_callbackqueryhandler.py#L252-L263), can you please review it ? 

Thanks